mend_sca_cleanup_tool/sca_cleanup_tool.py

```diff
@@ -58,35 +59,63 @@ def main():
         CONFIG = parse_args()
         
     setup_config()
+    if CONFIG.skip_report_generation:
+        print("Skip Report Generation Enabled - Reports will not be generated")
+    if CONFIG.skip_project_deletion:
+        print("Skip Project Deletion Enabled - Projects will not be deleted")
 
     if CONFIG.dry_run:
         print("Dry Run enabled - no reports or deletions will occur")
 
-    product_project_dict = get_projects_to_remove()
-    total_projects_to_delete = (sum([len(product_project_dict[x]) for x in product_project_dict]))
-    if not CONFIG.dry_run:
-        if total_projects_to_delete == 0:
-            print("No projects to clean up were found")
-            exit()
-        else:
-            print(f"Found {total_projects_to_delete} project(s) to delete, generating reports and removing project(s)...")
-        for product_token in product_project_dict:
-            for project in product_project_dict[product_token]:
-                if not CONFIG.skip_report_generation:
-                    try:
-                        generate_reports(project)
-                    except:
-                        print(f"There was an issue with the report generation, skipping deletion for project: {project['name']}")
-                        continue
-                else:
-                    print("skipReportGeneration flag found, skipping report generation")
-                if not CONFIG.skip_project_deletion:
-                    delete_scan(product_token, project)
-                else:
-                    print("skipProjectDeletion flag found, skipping project deletion")
+    processed_projects = {}
+    products = get_products()
+    with ThreadPoolExecutor(max_workers=CONFIG.project_parallelism_level) as executor:
+            futures = []
+            for product in products:
+                futures.append(executor.submit(process_product, product))
+                
+            for future in as_completed(futures):
+                processed_projects.update(future.result())
+    total_projects_to_delete = (sum([len(processed_projects[x]) for x in processed_projects]))   
+    if total_projects_to_delete == 0:
+        print("No projects to clean up were found")
+        exit()
+    elif CONFIG.dry_run:
+        print(f"Dry Run found {total_projects_to_delete} project(s) to delete")   
     else:
-        print(f"Dry Run found {total_projects_to_delete} project(s) to delete: {[project['name'] for projects in product_project_dict.values() for project in projects]}")
+        print(f"Deleted {total_projects_to_delete} project(s)")
+    if not CONFIG.skip_summary:
```

Comment:
``CONFIG.skip_summary`` is never actually configured when parsing the params.config file. Therefore this throws an error. The params.config file must set a default for this.